### PR TITLE
Port the `devServer` to the new Webpack config 

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Make `dotcom-rendering/configs/webpack/server.dev.mjs` work as a fully-working development server

## Why?

Follow-up on:
- #8777 
- #10033 

## Screenshots

`NODE_ENV=development pnpm webpack serve`

image

### CODE REVIEW

File: dotcom-rendering/configs/webpack/server.dev.mjs [hunk 1]
```diff
@@ -10,18 +10,114 @@
...
+import { success } from '../../../scripts/log.js';
+import { getContentFromURLMiddleware } from '../../src/server/lib/get-content-from-url.js';
 import { base } from './base.mjs';
 import { server } from './server.mjs';
 
+const DIRNAME = fileURLToPath(new URL('.', import.meta.url));
+
 /** @type {import("webpack").Configuration} */
-export const devServer = {
+export const serverDev = {
```
Comment (georgeblahblah) [inline dotcom-rendering/configs/webpack/server.dev.mjs:29]: 🔄

**Review by georgeblahblah — APPROVED** [review]

File: dotcom-rendering/configs/webpack/server.dev.mjs [hunk 2]
```diff
@@ -10,18 +10,114 @@
...
+	devServer: {
+		compress: false,
+		hot: false,
+		liveReload: true,
+		client: {
+			logging: 'warn',
+			overlay: true,
+		},
+		port: 3030,
```
Comment (arelra) [inline dotcom-rendering/configs/webpack/server.dev.mjs:38]: Could this be made a const to make it easier to spot and change if required?

**Review by arelra — COMMENTED** [review]
`make dev` for server, client.apps and client.web currently starts within ~8s

In your screenshot it is up to ~24s. Is this because all variants are being built? Is the normal dev build still approx ~8s?

File: dotcom-rendering/configs/webpack/server.dev.mjs (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/configs/webpack/server.dev.mjs:38]: Possibly, although the benefit of these webpack configs is that they are shorter and only have meaningful fields. If we extract each of the fields to const, then we are no further along in achieving prominence for any specific one.

We could move it to the top of the object if we feel that it’s more important?

**Comment by mxdvl** [comment]
> `make dev` for server, client.apps and client.web currently starts within ~8s
> 
> In your screenshot it is up to ~24s. Is this because all variants are being built? Is the normal dev build still approx ~8s?

Exactly: when only selecting the `client.web`, `client.apps` and `server.dev` we are standing at ~8s with this version as well.

File: dotcom-rendering/configs/webpack/server.dev.mjs (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/configs/webpack/server.dev.mjs]: I’ve worked around this, actually!

File: dotcom-rendering/configs/webpack/server.dev.mjs [hunk 3]
```diff
@@ -10,18 +10,114 @@
...
+			success(
+				[
+					'DEV server running on ',
+					'\x1b[36m', // cyan
+					'\x1b[4m', // underline
+					`http://localhost:${port}`,
+					'\x1b[0m', // reset
+				].join(''),
+			);
+		},
```
Comment (mxdvl) [inline dotcom-rendering/configs/webpack/server.dev.mjs:110]: This is new and replaces the inline log, which has the added benefit of actually only showing a clickable link once the server is actually listening.

**Review by arelra — APPROVED** [review]
✨

**Pull request merged**